samza-core/src/main/java/org/apache/samza/runtime/LocalApplicationRunner.java

```diff
@@ -179,20 +189,51 @@ public ApplicationStatus status(StreamApplication streamApp) {
   }
 
   /**
+   * Generates processorId for each {@link StreamProcessor} using the configured {@link ProcessorIdGenerator}
+   *
+   * @param config Application config
+   * @return String that uniquely represents an instance of {@link StreamProcessor} in the current JVM
+   */
+  /* package private */
+  String getProcessorId(Config config) {
```

Comment:
Summarizing our offline discussion with @vjagadish1989:

> However, if the interface for generating processorID is configurable, then it should ideally come from the user-facing layer.
> What is the purpose of getProcessorId in the JobCoordinator interface?

From the StreamProcessor's perspective, processorId is a way for it to know what part of the JobModel it needs to work on. Some JobCoordinators (e.g. ZK based JobCoordinator) may want to handle work assignment (and hence processorId assignment) dynamically as processors join and leave the group. Therefore, it makes sense to me to leave the option of  assigning the processorId dynamically at runtime open and let JC be the source of truth for these IDs. 

Regarding the ProcessorIdGenerator interface, it's pluggable in the sense that it lets JCs avoid reimplementing the same getProcessorId() logic (e.g. get id from config) by choosing a concrete implementation, but not in the sense that users can pick any one they like. A particular JobCoordinator implementation will usually require a specific ProcessorIdGenerator. Since we don't have that many JC implementations I don't mind removing it.

Does this make sense?